app/src/androidTest/java/com/github/onlynotesswent/ui/EndToEndTest.kt

```diff
@@ -227,17 +293,193 @@ class EndToEndTest {
     composeTestRule.onNodeWithTag("Save button").performClick()
 
     // Mock retrieval of notes
-    `when`(noteRepository.getRootNotesFrom(eq(testUser.uid), any(), any())).thenAnswer { invocation
+    `when`(noteRepository.getRootNotesFrom(eq(testUser1.uid), any(), any())).thenAnswer { invocation
       ->
       val onSuccess = invocation.getArgument<(List<Note>) -> Unit>(1)
       onSuccess(listOf(testNote))
     }
 
     // Trigger note retrieval and verify the notes are displayed
-    noteViewModel.getRootNotesFrom(testUser.uid)
+    noteViewModel.getRootNotesFrom(testUser1.uid)
     composeTestRule.onNodeWithTag("noteAndFolderList").assertIsDisplayed()
 
     // Verify that the note card is displayed
     composeTestRule.onNodeWithTag("noteCard").assertIsDisplayed()
   }
+
+  // Creates the mock behavior needed for the end-to-end flow of searching for testUser2 and viewing
+  // their profile and following them and then going to profile screen to unfollow and modify
+  // profile
+  private fun testEndToEndFlow2_init() {
```

Comment:
good modularisation of init function to keep end to end flow test clean and readable.
when we will need to add more end to end tests, we could combine all the mocks for all the inits in setup, as they will all probably be similar. for now this is not relevant.